packages/flutter/lib/src/widgets/animated_scroll_view.dart

```diff
@@ -83,51 +196,192 @@ class _ActiveItem implements Comparable<_ActiveItem> {
+  /// A delegate that controls the layout of the children within the
+  /// [AnimatedGrid].
+  ///
+  /// See also:
+  ///
+  ///  * [SliverGridDelegateWithFixedCrossAxisCount], which creates a layout with
+  ///    a fixed number of tiles in the cross axis.
+  ///  * [SliverGridDelegateWithMaxCrossAxisExtent], which creates a layout with
+  ///    tiles that have a maximum cross-axis extent.
+  /// {@endtemplate}
+  final SliverGridDelegate gridDelegate;
+
+  /// The state from the closest instance of this class that encloses the given
+  /// context.
+  ///
+  /// This method is typically used by [AnimatedGrid] item widgets that insert
+  /// or remove items in response to user input.
+  ///
+  /// If no [AnimatedGrid] surrounds the context given, then this function will
+  /// assert in debug mode and throw an exception in release mode.
+  ///
+  /// This method can be expensive (it walks the element tree).
+  ///
+  /// See also:
+  ///
+  ///  * [maybeOf], a similar function that will return null if no
+  ///    [AnimatedGrid] ancestor is found.
+  static AnimatedGridState of(BuildContext context) {
+    assert(context != null);
+    final AnimatedGridState? result = context.findAncestorStateOfType<AnimatedGridState>();
+    assert(() {
+      if (result == null) {
+        throw FlutterError.fromParts(<DiagnosticsNode>[
+          ErrorSummary('AnimatedGrid.of() called with a context that does not contain an AnimatedGrid.'),
+          ErrorDescription(
+            'No AnimatedGrid ancestor could be found starting from the context that was passed to AnimatedGrid.of().',
+          ),
+          ErrorHint(
+            'This can happen when the context provided is from the same StatefulWidget that '
+            'built the AnimatedGrid. Please see the AnimatedGrid documentation for examples '
+            'of how to refer to an AnimatedGridState object:\n'
+            '  https://api.flutter.dev/flutter/widgets/AnimatedGridState-class.html',
+          ),
+          context.describeElement('The context used was'),
+        ]);
+      }
+      return true;
+    }());
+    return result!;
+  }
+
+  /// The state from the closest instance of this class that encloses the given
+  /// context.
+  ///
+  /// This method is typically used by [AnimatedGrid] item widgets that insert
+  /// or remove items in response to user input.
+  ///
+  /// If no [AnimatedGrid] surrounds the context given, then this function will
+  /// return null.
+  ///
+  /// This method can be expensive (it walks the element tree).
+  ///
+  /// See also:
+  ///
+  ///  * [of], a similar function that will throw if no [AnimatedGrid] ancestor
+  ///    is found.
+  static AnimatedGridState? maybeOf(BuildContext context) {
+    assert(context != null);
+    return context.findAncestorStateOfType<AnimatedGridState>();
+  }
+
+  @override
+  AnimatedGridState createState() => AnimatedGridState();
+}
+
+/// The state for a scrolling container that animates items when they are
```

Comment:
We should probably have a `[AnimatedGrid]` link in this first line somewhere, so it's clear what widget it is associated with.